napari/utils/_proxies.py

```diff
@@ -73,6 +92,33 @@ def __getattr__(self, name: str):
             # )
         return self.create(super().__getattr__(name))
 
+    def __setattr__(self, name: str, value: Any):
+        if os.environ.get("NAPARI_ENSURE_PLUGIN_MAIN_THREAD", False):
```

Comment:
Is this something that could go into settings instead?
(e.g. how experimental octtree is configured)

I'm not a fan of having behavior deep into napari changed based on external configurations.  (But, I don't have a strong conviction here. :)

Reply:
> Is this something that could go into settings instead?

But this is for debugging purposes for the plugin creators. Not something that a common user may want to use.

Reply:
Checking documentation https://docs.python.org/3/library/os.html#os.environ, and it is a pure dict query. It will be faster than querying settings.